### PR TITLE
Add branch coverage to flutter test

### PR DESCRIPTION
**Pull request by liamappelbe**
Now that flutter is using package:coverage, adding branch coverage is just a matter of adding a command line flag (`--branch-coverage`) and plumbing it through to collect.

https://github.com/flutter/flutter/issues/108313

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/coverage_collector_test.dart [hunk 1]
```diff
@@ -2,8 +2,6 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-@Timeout.factor(10)
```
Comment (liamappelbe) [inline packages/flutter_tools/test/general.shard/coverage_collector_test.dart:5]: I added this in a previous PR and intended it to be temporary, but I forgot to remove it.

**Review by jonahwilliams — APPROVED** [review]
LGTM!

**Comment by jonahwilliams** [comment]
is there any disadvantage to always collecting branch coverage when coverage is true? Or defaulting to true?

**Comment by liamappelbe** [comment]
> is there any disadvantage to always collecting branch coverage when coverage is true? Or defaulting to true?

Branch coverage introduces a little more overhead. Going from no coverage to coverage is a much bigger overhead increase than coverage to branch coverage, but given how many complaints there already are about coverage being slow, I'd rather let the user opt into this.

**Comment by LongXiangGuo** [comment]
This case could not hit by the branch coverage
branch-coverage-issue

**Pull request merged**